lib/simctl.js

```diff
@@ -335,23 +339,39 @@ async function shutdown (udid) {
  * @param {string} runtimeId - Platform version, for example '10.3'.
  * @param {number} timeout [10000] - The maximum number of milliseconds to wait
  *                                   unit device creation is completed.
+ * @param {?SimDeviceOpts} opts - Simulator options for creating devives
  * @return {string} The UDID of the newly created device.
  * @throws {Error} If the corresponding simctl subcommand command
  *                 returns non-zero return code.
  */
-async function createDevice (name, deviceTypeId, runtimeId, timeout = 10000) {
+async function createDevice (name, deviceTypeId, runtimeId, timeout = 10000, opts = {}) {
+  const {
+    platform = 'iOS'
+  } = opts;
+
   let udid;
   // first make sure that the runtime id is the right one
   // in some versions of Xcode it will be a patch version
   try {
-    runtimeId = await getRuntimeForPlatformVersion(runtimeId);
+    runtimeId = await getRuntimeForPlatformVersion(runtimeId, platform);
   } catch (err) {
     log.warn(`Unable to find runtime for iOS '${runtimeId}'. Continuing`);
   }
 
-  log.debug(`Creating simulator with name '${name}', device type id '${deviceTypeId}' and runtime id '${runtimeId}'`);
+  const runtime = `${SIM_RUNTIME_NAME}${platform}-${runtimeId.replace('.', '-')}`;
   try {
-    const out = await simExec('create', 0, [name, deviceTypeId, runtimeId]);
+    let out;
```

Comment:
I would prefer to select runtime format based on the current SDK version rather than exception-controlled approach

Reply:
Yes, I also think so.
Maybe, https://github.com/appium/appium-xcode/blob/6ff33ae9dc849a921b3230bc4dec913ccab33b7d/lib/xcode.js#L148 can work for it... let me try.
(`xcodebuild -version -sdk` does not work in this case)

Reply:
https://github.com/appium/appium-ios-simulator/blob/de4938189bfe6f728d1328df4d141870132b53ab/lib/simulator.js#L31

Reply:
I mean if the Xcode CLI is over 10.2, simctl follows the 10.2 format even if we swith Xcode version with 10.1 by xcode-select

Reply:
you could probably just list all sdks and then check if any of them is >= 12.2 (`xcodebuild -showsdks`)

Reply:
Also, it would probably be nice to have a function like `isMinSDKInstalled(version, platform)` in appium-xcode

Reply:
feel free to use https://github.com/appium/appium-support/pull/107 for versions comparison

Reply:
I'm using it with getCommandLineToolsVersion and it's working well 👍 

`xcodebuild -showsdks` also depends on Xcode version which is specified by `xcode-select`. When I selected Xcode 10.1, then it showed 12.1. Xcode 10.2's case was 12.2, but `getCommandLineToolsVersion()` returned 10.2.0.xxxx in both case (so the simctl follows the new format)

Reply:
```
kazu$ xcodebuild -showsdks
iOS SDKs:
	iOS 12.1                      	-sdk iphoneos12.1

iOS Simulator SDKs:
	Simulator - iOS 12.1          	-sdk iphonesimulator12.1

macOS SDKs:
	macOS 10.14                   	-sdk macosx10.14

tvOS SDKs:
	tvOS 12.1                     	-sdk appletvos12.1

tvOS Simulator SDKs:
	Simulator - tvOS 12.1         	-sdk appletvsimulator12.1

watchOS SDKs:
	watchOS 5.1                   	-sdk watchos5.1

watchOS Simulator SDKs:
	Simulator - watchOS 5.1       	-sdk watchsimulator5.1

kazu$
kazu$ sudo xcode-select -s /Applications/Xcode.app/Contents/Developer
Password:
kazu$ xcodebuild -showsdks
iOS SDKs:
	iOS 12.2                      	-sdk iphoneos12.2

iOS Simulator SDKs:
	Simulator - iOS 12.2          	-sdk iphonesimulator12.2

macOS SDKs:
	macOS 10.14                   	-sdk macosx10.14

tvOS SDKs:
	tvOS 12.2                     	-sdk appletvos12.2

tvOS Simulator SDKs:
	Simulator - tvOS 12.2         	-sdk appletvsimulator12.2

watchOS SDKs:
	watchOS 5.2                   	-sdk watchos5.2

watchOS Simulator SDKs:
	Simulator - watchOS 5.2       	-sdk watchsimulator5.2
```